### PR TITLE
Add ESP-IDF framework support

### LINKED ISSUE #1691: Add ESP-IDF framework support
With #1687 it is functionally possible to support `target_os="espidf"` but unfortunately there are still compiler errors for things like missing socket constants and broken runtime behaviour.  ~This is a placeholder task to fix those issues (PR to follow #1687 landing)~

### PR DESCRIPTION
**Pull request by jasta**
Mostly fixing compiler errors but also working around nonblocking issues with accept, eventfd, etc.

Closes #1691

### CODE REVIEW

**Comment by jasta** [comment]
Note that this will conflict with #1687 but with the latest low-churn changes it only conflicts in one trivial place that I can easily fix in either this PR or the other one depending on which one lands first.

**Review by Thomasdezeeuw — APPROVED** [review]

File: src/sys/unix/net.rs [hunk 1]
```diff
@@ -44,18 +44,20 @@ pub(crate) fn new_socket(domain: libc::c_int, socket_type: libc::c_int) -> io::R
...
         target_os = "watchos",
+        target_os = "espidf",
     ))]
     {
         if let Err(err) = syscall!(fcntl(socket, libc::F_SETFL, libc::O_NONBLOCK)) {
             let _ = syscall!(close(socket));
             return Err(err);
         }
+        #[cfg(not(target_os = "espidf"))]
         if let Err(err) = syscall!(fcntl(socket, libc::F_SETFD, libc::FD_CLOEXEC)) {
```
[the remark this answers is not in the document]
Comment (jasta) [reply src/sys/unix/net.rs:61]: It is not relevant, there is no exec support for FreeRTOS which is what espidf runs.

**Review by Thomasdezeeuw — APPROVED** [review]

**Comment by Thomasdezeeuw** [comment]
I think this is good to be merged if you can resolve the merge conflict. Afterwards we can open an issue for official support, which would require a maintainer and CI support, etc.

**Review by Thomasdezeeuw — APPROVED** [review]

**Comment by Thomasdezeeuw** [comment]
Thanks @jasta

**Pull request merged**